app/models/granule.rb

```diff
@@ -107,6 +107,42 @@ def self.add_new_revision_to_granule(granule, current_user = User.find_by(role:
     end
   end
 
+  def self.ingest_specific(collection_concept_id, granule_concept_id, current_user = User.find_by(role: "admin"))
+    granule_info = Cmr.get_granule_with_collection_data(granule_concept_id)
+    collection = Collection.find_by(concept_id: granule_info["collection_concept_id"])
+
+    if granule_info["collection_concept_id"] != collection_concept_id
+      raise Cmr::CmrError.new, 'Granule record does not belong to this collection.'
+    end
+    if collection.nil?
+      raise Cmr::CmrError.new, 'Collection record associated with granule could not be found in system.'
```

Comment:
I don't know of a better alternative here at the moment, but instead of returning with something that indicates an error, you are raising an exception. I'm not sure if it needs to be changed, but it is propagating the need to use `rescue` in more places that we've discussed.

Reply:
Yeah in other languages that is kind of common, I don't think you want your return to be an error object or something.   Maybe this could be refactored to raise specific exception types (that is more common in other languages), but that might be a little overkill for this use case.